packer/plugin/provisioner.go

```diff
@@ -21,13 +21,13 @@ func (c *cmdProvisioner) Prepare(configs ...interface{}) error {
 	return c.p.Prepare(configs...)
 }
 
-func (c *cmdProvisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.Communicator) error {
+func (c *cmdProvisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.Communicator, generatedData interface{}) error {
```

Comment:
Howdy 🙂 , I spent a while reading this PR/RFC and there are some thoughts that I wanted to share here, sorry I didn't do that earlier !

`generatedData interface{}` sounds very untyped, could be anything and when a user will define/maintain provisioners plugins this is going to be very confusing ( I _think_ ). Basically we will know what this last parameter does based on it's _generally given name_; here: `generatedData` and/or docs.

I also saw that `generatedData` was mainly a `map[string]something` and I wanted to add that this would be the same as a stateBag; but we ( packer maintainer ) could make the usage of such a state bag better, safer and documented by owning and typing it. The underlying datastore could still be a map !

Was there something that made you go for these types ?

Reply:
Last question then I will stop 🙂 :

Do we want provisioners to be able to share infos too ? ex:

```go
    // If implemented; this Put would write to "provisioner-name.key"
    // we could add a special provisioner that will `Put` the result of a command into the store ?
    // It would be best if it was impossible to overwrite a value.
    err = env.Put(ctx, "key", /* value */ interface{}{})
    
    v, err := env.Get(ctx, "provisioner-name.key")
```

Reply:
I find difficult to understand why is called `generatedData` 🤔

Reply:
We may change the name, but essentially, information being passed through this struct is created/generated by the builder when it launches the instance, not provided by the user. This information primarily includes read-only connection information that Packer uses to connect to the instance (instance id, hostname, ssh/winrm passwords),